pipes/WDL/workflows/demux_deplete.wdl

```diff
@@ -116,7 +116,7 @@ workflow demux_deplete {
                 platform              = "ILLUMINA",
                 paired                = (illumina_demux.run_info[0]['indexes'] == '2'),
                 out_name              = "sra_metadata-~{flowcell_id}.tsv",
-                instrument_model      = select_first([instrument_model]),
+                instrument_model      = select_first(flatten([illumina_demux.instrument_model,[""]])),
```

Comment:
I'm a little confused what the `flatten` is for and why this can't be constructed a little simpler… (like `select_first([illumina_demux.instrument_model,""])`) and, actually, if we're going to keep the `instrument_model`) workflow level input, maybe the workflow level input should be optional (`String?`) and this line could become:
```
instrument_model = select_first([instrument_model, illumina_demux.instrument_model])
```
Meaning that a user-specified value would override, but absent that, it defaults to auto-inferring it.

Reply:
`illumina_demux.instrument_model` is an array as returned by the scatter-gather of demux jobs, no? I was under the impression we'd need to flatten it since [`select_first()`](https://github.com/openwdl/wdl/blob/main/versions/1.0/SPEC.md#x-select_firstarrayx) would still return an `Array[String]` in the proposed change. Or maybe I'm misunderstanding the WDL docs?